[2025-07-06T15:21:16.145+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-06T15:21:16.179+0000] {taskinstance.py:2631} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: process_sword_messages.process_sword_messages manual__2025-07-06T15:16:06.617316+00:00 [queued]>
[2025-07-06T15:21:16.189+0000] {taskinstance.py:2631} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: process_sword_messages.process_sword_messages manual__2025-07-06T15:16:06.617316+00:00 [queued]>
[2025-07-06T15:21:16.189+0000] {taskinstance.py:2884} INFO - Starting attempt 2 of 2
[2025-07-06T15:21:16.247+0000] {taskinstance.py:2907} INFO - Executing <Task(SparkSubmitOperator): process_sword_messages> on 2025-07-06 15:16:06.617316+00:00
[2025-07-06T15:21:16.254+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=585) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-06T15:21:16.255+0000] {standard_task_runner.py:72} INFO - Started process 586 to run task
[2025-07-06T15:21:16.254+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'process_sword_messages', 'process_sword_messages', 'manual__2025-07-06T15:16:06.617316+00:00', '--job-id', '57', '--raw', '--subdir', 'DAGS_FOLDER/process_sword_messages.py', '--cfg-path', '/tmp/tmp0sit_8t5']
[2025-07-06T15:21:16.256+0000] {standard_task_runner.py:105} INFO - Job 57: Subtask process_sword_messages
[2025-07-06T15:21:16.443+0000] {task_command.py:467} INFO - Running <TaskInstance: process_sword_messages.process_sword_messages manual__2025-07-06T15:16:06.617316+00:00 [running]> on host 87943ecf3d0d
[2025-07-06T15:21:16.588+0000] {taskinstance.py:3157} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='process_sword_messages' AIRFLOW_CTX_TASK_ID='process_sword_messages' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T15:16:06.617316+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-06T15:16:06.617316+00:00'
[2025-07-06T15:21:16.589+0000] {taskinstance.py:740} INFO - ::endgroup::
[2025-07-06T15:21:16.624+0000] {base.py:84} INFO - Retrieving connection 'spark-ebsim'
[2025-07-06T15:21:16.626+0000] {spark_submit.py:474} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin
[2025-07-06T15:21:20.040+0000] {spark_submit.py:644} INFO - Error importing protobuf schema: No module named 'protobuf_schemas'
[2025-07-06T15:21:20.040+0000] {spark_submit.py:644} INFO - Traceback (most recent call last):
[2025-07-06T15:21:20.041+0000] {spark_submit.py:644} INFO - File "/opt/bitnami/spark/jobs/sword_processor_job.py", line 9, in <module>
[2025-07-06T15:21:20.044+0000] {spark_submit.py:644} INFO - from protobuf_schemas.simulation_client_pb2 import SimToClient
[2025-07-06T15:21:20.044+0000] {spark_submit.py:644} INFO - ModuleNotFoundError: No module named 'protobuf_schemas'
[2025-07-06T15:21:20.094+0000] {spark_submit.py:644} INFO - 25/07/06 15:21:20 INFO ShutdownHookManager: Shutdown hook called
[2025-07-06T15:21:20.097+0000] {spark_submit.py:644} INFO - 25/07/06 15:21:20 INFO ShutdownHookManager: Deleting directory /tmp/spark-ebc48be9-f91d-4fd0-aee0-848ae302408d
[2025-07-06T15:21:20.150+0000] {taskinstance.py:3336} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 776, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 742, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin. Error code is: 1.
[2025-07-06T15:21:20.156+0000] {taskinstance.py:1242} INFO - Marking task as FAILED. dag_id=process_sword_messages, task_id=process_sword_messages, run_id=manual__2025-07-06T15:16:06.617316+00:00, execution_date=20250706T151606, start_date=20250706T152116, end_date=20250706T152120
[2025-07-06T15:21:20.323+0000] {taskinstance.py:349} INFO - ::group::Post task execution logs
[2025-07-06T15:21:20.325+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 57 for task process_sword_messages (Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin. Error code is: 1.; 586)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3029, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 282, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3184, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3208, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 776, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 742, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 197, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 566, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --deploy-mode client /opt/bitnami/spark/jobs/sword_processor_job.py --input-file /opt/bitnami/spark/jobs/test_sword_messages.bin. Error code is: 1.
[2025-07-06T15:21:20.346+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-07-06T15:21:20.387+0000] {taskinstance.py:3924} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-06T15:21:20.432+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
